rust/lance-index/benches/4bitpq_dist_table.rs

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+//! Benchmark of building PQ distance table.
+
+use std::iter::repeat;
+
+use arrow_array::types::Float32Type;
+use arrow_array::{FixedSizeListArray, UInt8Array};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use lance_arrow::FixedSizeListArrayExt;
+use lance_index::vector::pq::ProductQuantizer;
+use lance_linalg::distance::DistanceType;
+use lance_testing::datagen::generate_random_array_with_seed;
+use rand::{prelude::StdRng, Rng, SeedableRng};
+
+#[cfg(target_os = "linux")]
+use pprof::criterion::{Output, PProfProfiler};
+
+const PQ: usize = 96;
+const DIM: usize = 1536;
+const TOTAL: usize = 5 * 1024 * 1024;
```

Comment:
Lets all use more realistic numbers?, i.e., 8K - 16K, so that we can measure other piece i.e., table construction, and code transpose more realistically.

Reply:
make sense, done